src/GafferScene/LightToCamera.cpp

```diff
@@ -0,0 +1,312 @@
+		{
+			return parameterData->readable();
+		}
+
+		return defaultValue;
+	}
+
+	// If the attributes contain a light, return a corresponding camera
+	// Just returns the first light it finds, which could produce inconsistent results if you
+	// have multiple contradictory lights in the same location.  This seems like a case that should
+	// be avoided anyway
+	IECore::CameraPtr convertLightToCamera( IECore::ConstCompoundObjectPtr attributes, int resolution )
+	{
+		for( IECore::CompoundObject::ObjectMap::const_iterator it = attributes->members().begin();
+			it != attributes->members().end(); it++ )
+		{
+			const std::string &attributeName = it->first.string();
+			if( !( boost::ends_with( attributeName, ":light" ) || attributeName == "light" ) )
+			{
+				continue;
+			}
+
+			const IECore::ObjectVector *shaderVector = IECore::runTimeCast<const IECore::ObjectVector>( it->second.get() );
+			if( !shaderVector || shaderVector->members().empty() )
+			{
+				continue;
+			}
+
+			IECore::InternedString shaderName;
+			const IECore::Shader *shader = IECore::runTimeCast<const IECore::Shader>( shaderVector->members().back().get() );
+			if( !shader )
+			{
+				continue;
+			}
+
+
+			std::string metadataTarget = attributeName + ":" + shader->getName();
+			const IECore::CompoundData *shaderParameters = shader->parametersData();
+			ConstStringDataPtr type = Metadata::value<StringData>( metadataTarget, "type" );
+
+			if( !type || !shaderParameters )
+			{
+				continue;
+			}
+
+			IECore::CameraPtr result = new IECore::Camera();
+
+
+			float screenWindowScale = 1.0f;
+			if( type->readable() == "distant" )
+			{
+			    const float locatorScale = parameter<float>( metadataTarget, shaderParameters, "locatorScaleParameter", 1 );
+				
+				result->parameters()["projection"] = new StringData( "orthographic" );
+				result->parameters()["clippingPlanes"] = new V2fData( V2f( -100000, 100000 ) );
+				screenWindowScale = locatorScale;
+			}
+			else if( type->readable() == "spot" )
+			{
+
+				/// \todo - this logic should look much cleaner once we have LightAccessor
+				float coneAngle = parameter<float>( metadataTarget, shaderParameters, "coneAngleParameter", 0.0f );
+				float penumbraAngle = parameter<float>( metadataTarget, shaderParameters, "penumbraAngleParameter", 0.0f );
+
+				if( ConstStringDataPtr angleUnit = Metadata::value<StringData>( metadataTarget, "angleUnit" ) )
+				{
+					if( angleUnit->readable() == "radians" )
+					{
+						coneAngle *= 180.0 / M_PI;
+						penumbraAngle *= 180 / M_PI;
+					}
+				}
+
+				const std::string *penumbraType = NULL;
+				ConstStringDataPtr penumbraTypeData = Metadata::value<StringData>( metadataTarget, "penumbraType" );
+				if( penumbraTypeData )
+				{
+					penumbraType = &penumbraTypeData->readable();
+				}
+
+				float outerAngle = coneAngle;
+				if( penumbraType && *penumbraType == "outset" )
```

Comment:
Currently, I match the fov to the outer angle, and both the inset and absolute types set outerAngle = coneAngle.  Can add comment to this effect.